flink-yarn/src/test/java/org/apache/flink/yarn/YarnClusterDescriptorTest.java

```diff
@@ -854,6 +855,25 @@ public void testGenerateApplicationMasterEnv() throws IOException {
         assertEquals(flinkHomeDir.getPath(), masterEnv.get(YarnConfigKeys.ENV_CLIENT_HOME_DIR));
     }
 
+    @Test
+    public void testEnvFlinkLibDirVarNotOverriddenBySystem() throws IOException {
```

Comment:
I think what we need to test is `ENV_FLINK_LIB_DIR` is not overridden by user specified envs via `CONTAINERIZED_MASTER_ENV_PREFIX`, not overridden by system environment.

```
flinkConfig.setString(
                ResourceManagerOptions.CONTAINERIZED_MASTER_ENV_PREFIX
                        + ConfigConstants.ENV_FLINK_LIB_DIR,
                "/path/of/flink-lib");
```

Reply:
`YarnClusterDescriptor` is running on the client side. So I do not think the current test make sense if you really want to test system env on the JM/TM side should not override `ENV_FLINK_LIB_DIR`.

Reply:
I see your point. I agree current test is useless as it is run in client side. Using `CONTAINERIZED_MASTER_ENV_PREFIX` makes more sense.